sentry_sdk/integrations/starlette.py

```diff
@@ -104,58 +104,54 @@ def _enable_span_for_middleware(middleware_class):
 
     async def _create_span_call(app, scope, receive, send, **kwargs):
         # type: (Any, Dict[str, Any], Callable[[], Awaitable[Dict[str, Any]]], Callable[[Dict[str, Any]], Awaitable[None]], Any) -> None
-        hub = Hub.current
-        integration = hub.get_integration(StarletteIntegration)
-        if integration is not None:
-            middleware_name = app.__class__.__name__
-
-            # Update transaction name with middleware name
-            name, source = _get_transaction_from_middleware(app, scope, integration)
-            if name is not None:
-                Scope.get_current_scope().set_transaction_name(
-                    name,
-                    source=source,
-                )
+        integration = sentry_sdk.get_client().get_integration(StarletteIntegration)
+        if integration is None:
+            return await old_call(app, scope, receive, send, **kwargs)
```

Comment:
We could use `ensure_integration_enabled_async` here

Reply:
I was using it initially but it seems there's something wrong with it. When I [removed](https://github.com/getsentry/sentry-python/pull/2877/commits/25fdff1ad903942ada1a69d4e94c59b4a51da65a) it the tests started passing. So we should probably fix the async decorator. (The sync one works fine.)

Reply:
https://github.com/getsentry/sentry-python/issues/2892

Reply:
Ok, I guess than it is ok.